### PR TITLE
UF-3071: Create integration for updating status of a case.

### PR DESCRIPTION
**Pull request by MickeWallner**

### CODE REVIEW

File: src/main/java/se/sundsvall/processengine/parkingpermit/integration/casedata/CaseDataClient.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+	@PatchMapping(path = "/errands/{caseId}/decisions", produces = APPLICATION_JSON_VALUE, consumes = APPLICATION_JSON_VALUE)
+	ResponseEntity<Void> patchDecision(@PathVariable(name = "caseId") String caseId, @RequestBody DecisionDTO decisionDTO);
+
+	/**
+	 * Gets an errand by id.
+	 *
+	 * @param errandId of errand to get
+	 * @throws org.zalando.problem.ThrowableProblem
+	 */
+	@GetMapping(path = "/errands/{errandId}", produces = APPLICATION_JSON_VALUE, consumes = APPLICATION_JSON_VALUE)
```
Comment (foobar-1337) [inline src/main/java/se/sundsvall/processengine/parkingpermit/integration/casedata/CaseDataClient.java:39]: This method doesn't consume json

File: src/main/java/se/sundsvall/processengine/parkingpermit/integration/casedata/configuration/CaseDataConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+import org.springframework.cloud.openfeign.FeignBuilderCustomizer;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Import;
+import org.springframework.security.oauth2.client.registration.ClientRegistrationRepository;
+import se.sundsvall.dept44.configuration.feign.FeignConfiguration;
+import se.sundsvall.dept44.configuration.feign.FeignMultiCustomizer;
+import se.sundsvall.dept44.configuration.feign.decoder.ProblemErrorDecoder;
+
+
```
Comment (foobar-1337) [inline src/main/java/se/sundsvall/processengine/parkingpermit/integration/casedata/configuration/CaseDataConfiguration.java:11]: Excessive use of newlines.

File: src/main/java/se/sundsvall/processengine/parkingpermit/integration/camunda/CaseDataWorker.java [hunk 3]
```diff
@@ -1,32 +1,34 @@
...
+import java.util.Map;
+
+import static generated.se.sundsvall.casedata.StakeholderDTO.RolesEnum.APPLICANT;
+import static generated.se.sundsvall.casedata.StakeholderDTO.TypeEnum.PERSON;
+import static java.util.Collections.emptyList;
+import static java.util.Optional.ofNullable;
 
 @Component
-@ExternalTaskSubscription("caseData")
+//@ExternalTaskSubscription("caseData")
```
Comment (MattiasGradinApendo) [inline src/main/java/se/sundsvall/processengine/parkingpermit/integration/camunda/CaseDataWorker.java:23]: You forgot to uncomment this line

File: src/test/java/se/sundsvall/processengine/parkingpermit/integration/camunda/CaseDataWorkerTest.java [hunk 4]
```diff
@@ -6,28 +6,23 @@
...
 
-import static org.junit.jupiter.api.Assertions.*;
+import static generated.se.sundsvall.casedata.StakeholderDTO.RolesEnum.ADMINISTRATOR;
 import static org.mockito.Mockito.*;
 
 @ExtendWith(MockitoExtension.class)
 class CaseDataWorkerTest {
     @Mock
-    ErrandsClient mockErrandClient;
+    CaseDataClient mockCaseDataClient;
```
Comment (foobar-1337) [inline src/test/java/se/sundsvall/processengine/parkingpermit/integration/camunda/CaseDataWorkerTest.java:25]: private?

File: src/test/java/se/sundsvall/processengine/parkingpermit/integration/camunda/CaseDataWorkerTest.java [hunk 5]
```diff
@@ -6,28 +6,23 @@
...
     @Mock
     CitizenClient mockCitizenClient;
```
Comment (foobar-1337) [inline src/test/java/se/sundsvall/processengine/parkingpermit/integration/camunda/CaseDataWorkerTest.java:27]: private?

File: src/test/java/se/sundsvall/processengine/parkingpermit/integration/camunda/DecisionWorkerTest.java [hunk 6]
```diff
@@ -1,42 +1,35 @@
...
-import se.sundsvall.processengine.parkingpermit.integration.exceptions.ServiceException;
+import se.sundsvall.processengine.parkingpermit.integration.casedata.CaseDataClient;
 import se.sundsvall.processengine.parkingpermit.testutils.TestDataCreator;
 
-import static org.junit.jupiter.api.Assertions.*;
 import static org.mockito.Mockito.*;
 
 @ExtendWith(MockitoExtension.class)
 class DecisionWorkerTest {
     DecisionWorker decisionWorker;
```
Comment (foobar-1337) [inline src/test/java/se/sundsvall/processengine/parkingpermit/integration/camunda/DecisionWorkerTest.java:22]: private?

(comment applicable for all member-fields in this class)

File: src/main/java/se/sundsvall/processengine/parkingpermit/integration/camunda/CaseDataWorker.java (hunk 3 above)
Comment (foobar-1337) [inline src/main/java/se/sundsvall/processengine/parkingpermit/integration/camunda/CaseDataWorker.java:23]: Why is this out-commented?
Remove instead?

Should it be "activated" again later on? Maybe add a "TODO" as well?

File: src/main/java/se/sundsvall/processengine/parkingpermit/integration/camunda/DecisionWorker.java [hunk 7]
```diff
@@ -1,36 +1,34 @@
...
 import java.time.ZoneOffset;
 import java.util.Optional;
 
+import static generated.se.sundsvall.casedata.DecisionDTO.DecisionTypeEnum.APPROVAL;
+import static generated.se.sundsvall.casedata.DecisionDTO.DecisionTypeEnum.REJECTION;
+import static java.util.Optional.ofNullable;
+
 @Component
-@ExternalTaskSubscription("updateDecision")
+//@ExternalTaskSubscription("updateDecision")
```
Comment (foobar-1337) [inline src/main/java/se/sundsvall/processengine/parkingpermit/integration/camunda/DecisionWorker.java:24]: Why?

Should it be "activated" again later on? Maybe add a "TODO" as well?

File: src/main/java/se/sundsvall/processengine/parkingpermit/integration/camunda/DecisionWorker.java [hunk 8]
```diff
@@ -42,34 +40,34 @@ public void execute(ExternalTask externalTask, ExternalTaskService externalTaskS
...
+            ZoneOffset zoneOffSet= ZoneId.systemDefault().getRules().getOffset(LocalDateTime.now());
             OffsetDateTime offsetDateTime = OffsetDateTime.now(zoneOffSet);
             decision.setDecidedAt(offsetDateTime);
         }
         else {
-            decision.setDecisionType(DecisionType.REJECTION);
+            decision.setDecisionType(REJECTION);
             decision.setDescription("Personen är inte skriven i Sundsvalls kommun. Rekommenderar därför avslag på ansökan.");
-            ZoneOffset zoneOffSet= ZoneOffset.of("+02:00");
+            ZoneOffset zoneOffSet= ZoneId.systemDefault().getRules().getOffset(LocalDateTime.now());
```
Comment (foobar-1337) [inline src/main/java/se/sundsvall/processengine/parkingpermit/integration/camunda/DecisionWorker.java:64]: We have a util-method for this in dept44.

se.sundsvall.dept44.util.DateUtils.toOffsetDateTimeWithLocalOffset(OffsetDateTime date)

**Review by MattiasGradinApendo — CHANGES_REQUESTED** [review]
Looks good, just fix the uncommented lines :)

File: src/main/java/se/sundsvall/processengine/parkingpermit/integration/camunda/CaseDataWorker.java (hunk 3 above)
Comment (MickeWallner) [reply src/main/java/se/sundsvall/processengine/parkingpermit/integration/camunda/CaseDataWorker.java]: Fixed

File: src/main/java/se/sundsvall/processengine/parkingpermit/integration/casedata/CaseDataClient.java (hunk 1 above)
Comment (MickeWallner) [reply src/main/java/se/sundsvall/processengine/parkingpermit/integration/casedata/CaseDataClient.java]: Fixed

File: src/main/java/se/sundsvall/processengine/parkingpermit/integration/casedata/configuration/CaseDataConfiguration.java (hunk 2 above)
Comment (MickeWallner) [reply src/main/java/se/sundsvall/processengine/parkingpermit/integration/casedata/configuration/CaseDataConfiguration.java]: Fixed

File: src/test/java/se/sundsvall/processengine/parkingpermit/integration/camunda/CaseDataWorkerTest.java (hunk 4 above)
Comment (MickeWallner) [reply src/test/java/se/sundsvall/processengine/parkingpermit/integration/camunda/CaseDataWorkerTest.java]: Fixed

File: src/main/java/se/sundsvall/processengine/parkingpermit/integration/camunda/CaseDataWorker.java (hunk 3 above)
Comment (MickeWallner) [reply src/main/java/se/sundsvall/processengine/parkingpermit/integration/camunda/CaseDataWorker.java]: Just for be able to start locally. Forgot to uncomment. Fixed.

File: src/main/java/se/sundsvall/processengine/parkingpermit/integration/camunda/DecisionWorker.java (hunk 8 above)
Comment (MickeWallner) [reply src/main/java/se/sundsvall/processengine/parkingpermit/integration/camunda/DecisionWorker.java]: Fixed.

**Review by MattiasGradinApendo — APPROVED** [review]

**Pull request merged**